src/OhmmsSoA/VectorSoaContainer.h

```diff
@@ -119,41 +113,51 @@ struct VectorSoaContainer
     return *this;
   }
 
-  ///initialize the data members
-  __forceinline void setDefaults()
-  {
-    nLocal     = 0;
-    nGhosts    = 0;
-    nAllocated = 0;
-    myData     = nullptr;
-  }
-
   /** resize myData
        * @param n nLocal
        *
        * nAllocated is used to ensure no memory leak
        */
   __forceinline void resize(size_t n)
   {
-    static_assert(std::is_same<Element_t, typename Alloc::value_type>::value, "VectorSoaContainer and Alloc data types must agree!");
+    static_assert(std::is_same<Element_t, typename Alloc::value_type>::value,
+                  "VectorSoaContainer and Alloc data types must agree!");
+    if (n == 0)
+    {
+      free();
+      return;
```

Comment:
It looks like the code was restructured to use if/else, but the returns were not removed?